extras/rya.export/export.accumulo/src/main/java/org/apache/rya/export/accumulo/AccumuloRyaStatementStore.java

```diff
@@ -0,0 +1,225 @@
+ */
+package org.apache.rya.export.accumulo;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.log4j.Logger;
+import org.apache.rya.export.accumulo.common.InstanceType;
+import org.apache.rya.export.accumulo.util.AccumuloInstanceDriver;
+import org.apache.rya.export.accumulo.util.AccumuloRyaUtils;
+import org.apache.rya.export.api.MergerException;
+import org.apache.rya.export.api.conf.AccumuloMergeConfiguration;
+import org.apache.rya.export.api.store.AddStatementException;
+import org.apache.rya.export.api.store.ContainsStatementException;
+import org.apache.rya.export.api.store.FetchStatementException;
+import org.apache.rya.export.api.store.RemoveStatementException;
+import org.apache.rya.export.api.store.RyaStatementStore;
+import org.apache.rya.export.api.store.UpdateStatementException;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+
+import info.aduna.iteration.CloseableIteration;
+import mvm.rya.accumulo.AccumuloRyaDAO;
+import mvm.rya.api.RdfCloudTripleStoreConstants;
+import mvm.rya.api.domain.RyaStatement;
+import mvm.rya.api.persist.RyaDAOException;
+import mvm.rya.api.resolver.RyaTripleContext;
+import mvm.rya.api.resolver.triple.TripleRowResolverException;
+
+/**
+ * Allows specific CRUD operations an Accumulo {@link RyaStatement} storage
+ * system.
+ * <p>
+ * The operations specifically:
+ * <li>fetch all rya statements in the store</li>
+ * <li>add a rya statement to the store</li>
+ * <li>remove a rya statement from the store</li>
+ * <li>update an existing rya statement with a new one</li>
+ *
+ * One would use this {@link AccumuloRyaStatementStore} when they have an
+ * Accumulo database that is used when merging in data or exporting data.
+ */
+public class AccumuloRyaStatementStore implements RyaStatementStore {
+    private static final Logger log = Logger.getLogger(AccumuloRyaStatementStore.class);
+
+    private final AccumuloRyaDAO accumuloRyaDao;
+    private final String tablePrefix;
+    private final Set<IteratorSetting> iteratorSettings = new HashSet<>();
+    private final AccumuloInstanceDriver accumuloInstanceDriver;
+
+    /**
+     * Creates a new instance of {@link AccumuloRyaStatementStore}.
+     * @param config the {@link AccumuloMergeConfiguration}. (not {@code null})
+     * @param isParent {@code true} if the {@link AccumuloRyaStatementStore} is
+     * the parent instance.  {@code false} if it is the child instance.
+     * @throws MergerException
+     */
+    public AccumuloRyaStatementStore(final AccumuloMergeConfiguration config, final boolean isParent) throws MergerException {
```

Comment:
don't like the use of a flag to change the constructor here.  Instead of passing in the config, just pass in the 6 required fields.  The RyaStatementStore really shouldn't have an idea of parent/child relationships.  It should just care about statements